### PR TITLE
odb: refactor cpp tests

### PR DESCRIPTION
**Pull request by fgaray**
Some refactors to support subfolders for other projects inside cpp for odb (like scan) and also fixed clang-tidy warnings in the tests.

### CODE REVIEW

File: src/odb/test/cpp/helper/env.cpp
Comment (maliberty) [inline src/odb/test/cpp/helper/env.cpp:1]: missing copyright  header

File: src/odb/test/cpp/helper/env.h
Comment (maliberty) [inline src/odb/test/cpp/helper/env.h:1]: missing copyright  header

File: src/odb/test/cpp/helper/helper.cpp
Comment (maliberty) [inline src/odb/test/cpp/helper/helper.cpp:1]: missing copyright  header

File: src/odb/test/cpp/helper/helper.h
Comment (maliberty) [inline src/odb/test/cpp/helper/helper.h:1]: missing copyright header

File: src/odb/test/cpp_tests.tcl [hunk 1]
```diff
@@ -2,9 +2,9 @@ source "helpers.tcl"
 
 set test_dir [pwd]
 set openroad_dir [file dirname [file dirname [file dirname $test_dir]]]
-set tests_path [file join $openroad_dir "build" "src" "odb" "test" "cpp"]
+set tests_path [file join $openroad_dir "build" "src" "odb" "test" "cpp" ""]
```
Comment (maliberty) [inline src/odb/test/cpp_tests.tcl:5]: why the extra ""?

**Review by maliberty — CHANGES_REQUESTED** [review]
Thanks - the odb test can use some attention

File: src/odb/test/cpp_tests.tcl (hunk 1 above)
Comment (fgaray) [reply src/odb/test/cpp_tests.tcl]: It may have been there from my previous attempt. I removed it.

Thanks Matt for catching this!

**Comment by fgaray** [comment]
Hi @maliberty , are there any pending changes that I should take care in this PR? Thanks!

File: src/odb/test/cpp/helper/env.h [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+// INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+// CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+// ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+// POSSIBILITY OF SUCH DAMAGE.
+
+#include <string>
+
+#pragma once
+
+std::string testTmpPath(const std::string& file);
```
Comment (maliberty) [inline src/odb/test/cpp/helper/env.h:37]: odb namespace?

Comment (fgaray) [reply src/odb/test/cpp/helper/env.h:38]: Makes sense! thanks!

**Review by maliberty — APPROVED** [review]

**Pull request merged**